assets/javascripts/modules/autoComplete.js

```diff
@@ -314,28 +314,104 @@ var suggestionsEvent = function () {
   });
 };
 
+
+/**
+  * Generate random variable name to ensure no conflicts
+*/
+var randomVariableName = function () {
+  var text = "suggestions";
+  var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  for( var i=0; i < 5; i++ ) {
+      text += possible.charAt(Math.floor(Math.random() * possible.length));
+  }
+  return text;
+};
+
+/**
+  * Either grab a given global variable or generate one from a given select input
+*/
+var getSuggestions = function (el) {
+  var suggestions;
+  var dataLinked = $('select[id="' + el.attr("data-linked") + '"]');
+  if(dataLinked.length > 0){
+    // using a select input in-page as the source for the suggestions
+    // this is useful when using a select as a fallback for the autocomplete
+    suggestions = [];
+    dataLinked.find('option').each(function(){
+      if($(this).text() > ""){
+        suggestions.push({"title":$(this).text(), "value":$(this).attr("value")});
+      }
+    });
+    // need to populate the data-suggestions attribute from this for validation
+    // generate a global variable to hold the data and insert into data-suggestions attribute
+    var r = randomVariableName();
+    window[r] = suggestions;
```

Comment:
As the validator expects a global. Agreed this could be changed but it is outside the scope of this PR and can be picked up in a separate one.